### PR TITLE
GeoIPbackend improvements

### LINKED ISSUE #5268: GeoIP backend should support libmaxminddb
- Program: Authoritative
 - Issue type: Feature request

### Short description
To support the new GeoIP2 databases, libmaxminddb support should be added to geoipbackend

### PR DESCRIPTION
**Pull request by cmouse**
### Short description
- Adds MMDB support. Now geoip backend can be compiled without geoip support. Closes #5268 
- Adds location support
- Fixes SERVFAIL if expansion is empty

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: m4/pdns_with_geo.m4 [hunk 1]
```diff
@@ -1,7 +1,46 @@
 AC_DEFUN([PDNS_CHECK_GEOIP], [
-  PKG_CHECK_MODULES([GEOIP], [geoip],[],
-    AC_MSG_ERROR([Could not find libGeoIP])
+  PKG_CHECK_MODULES([GEOIP], [geoip], AC_DEFINE([HAVE_GEOIP], [1], [Define this if you have GeoIP]), [have_geoip=0])
+  AC_ARG_WITH([maxminddb_incdir],
```
Comment (pieterlexis) [inline m4/pdns_with_geo.m4]: libmaxminddb has pkgconfig support: https://github.com/maxmind/libmaxminddb/blob/master/src/libmaxminddb.pc.in
But this is not in the Ubuntu Artful package :(

File: docs/backends/geoip.rst [hunk 2]
```diff
@@ -46,15 +50,36 @@ defaults suite you.
 ``geoip-database-files``
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. warning::
```
Comment (pieterlexis) [inline docs/backends/geoip.rst]:
```rst
.. deprecated:: 4.2.0
  This setting has been removed
```

File: m4/pdns_with_geo.m4 (hunk 1 above)
Comment (cmouse) [reply m4/pdns_with_geo.m4]: yes, i know. I can add pkgconfig attempt first.

Comment (zeha) [reply m4/pdns_with_geo.m4]: artful has it: https://packages.ubuntu.com/artful/amd64/libmaxminddb-dev/filelist

Comment (cmouse) [reply m4/pdns_with_geo.m4]: in fact, if you look few lines forward, you'll notice that there is PKG_CHECK_MODULES for MMDB

Comment (cmouse) [reply m4/pdns_with_geo.m4]: @pieterlexis ?

Comment (pieterlexis) [reply m4/pdns_with_geo.m4]: :+1:

**Pull request merged**